specification/resourcegraph/resource-manager/Microsoft.ResourceGraph/preview/2018-09-01-preview/resourcegraph.json

```diff
@@ -411,6 +495,142 @@
         }
       ]
     },
+    "ResourceChangesRequestParameters": {
+      "description": "The parameters for a specific changes request.",
+      "type": "object",
+      "properties": {
+        "resourceId": {
+          "description": "Specifies the resource for a changes request.",
+          "type": "string"
+        },
+        "interval": {
+          "description": "Specifies the date and time interval for a changes request.",
+          "allOf": [
+            {
+              "$ref": "#/definitions/DateTimeInterval"
+            }
+          ]
+        },
+        "$skipToken": {
+          "description": "Acts as the continuation token for paged responses.",
+          "type": "string"
+        },
+        "$top": {
+          "description": "The maximum number of changes the client can accept in a paged response.",
+          "type": "integer",
+          "format": "int32",
+          "minimum": 1,
+          "maximum": 1000
+        }
+      },
+      "required": [
+        "resourceId",
+        "interval"
+      ]
+    },
+    "ResourceChangeList": {
+      "description": "A list of changes associated with a resource over a specific time interval.",
+      "type": "object",
+      "properties": {
+        "changes": {
+          "description": "The pageable value returned by the operation, i.e. a list of changes to the resource.\n\n- The list is ordered from the most recent changes to the least recent changes.\n- This list will be empty if there were no changes during the requested interval.\n- The `Before` snapshot timestamp value of the oldest change can be outside of the specified time interval.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ResourceChangeData"
+          }
+        },
+        "$skipToken": {
+          "description": "Skip token that encodes the skip information while executing the current request"
+        }
+      }
+    },
+    "ResourceChangeData": {
+      "description": "Data on a specific change, represented by a pair of before and after resource snapshots.",
+      "type": "object",
+      "properties": {
+        "changeId": {
+          "description": "The change ID. Valid and unique within the specified resource only.",
+          "type": "string"
+        },
+        "beforeSnapshot": {
```

Comment:
Have you considered exposing calculated deltas? I.e. only enumerate changed properties?

Reply:
Yes, we are planning to do this eventually. However, this will take time, and we want to release the currently proposed API because it already is quite valuable.